datafusion/core/src/datasource/physical_plan/parquet/mod.rs

```diff
@@ -586,7 +586,7 @@ impl ParquetExec {
         )
     }
 
-    fn with_file_groups(mut self, file_groups: Vec<Vec<PartitionedFile>>) -> Self {
+    pub fn with_file_groups(mut self, file_groups: Vec<Vec<PartitionedFile>>) -> Self {
```

Comment:
I think it will be easier to provide this

```rust
   let builder = ParquetExecBuilder::builder_from_parquet_exec(parquet_exec)
         .with_file_groups(...)
         .build()
```

The `builder_from_parquet_exec` is a new function

Reply:
I agree -- creating a builder from an existing ParquetExec would be helpful. I will try and make a PR for this

Reply:
The more I looked into this the more I think `with_file_groups` should not be public (as it does not recompute all required properties, such as the `Statistics` for example)

Reply:
PR with an alternate proposal: https://github.com/apache/datafusion/pull/12799